modules/kubernetes/src/main/java/org/apache/ignite/kubernetes/KubernetesConnectorConfigurator.java

```diff
@@ -0,0 +1,63 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.ignite.kubernetes;
+
+/**
+ * Interface provide methods to configure Kubernetes connection.
+ */
+public interface KubernetesConnectorConfigurator {
```

Comment:
@ptupitsyn Connection to k8s API is independent on Ignite node type. IpFinder and AddressFinder both have the same setters to configure the connection (namespace, service, etc). IMHO the right implementation would be to create a class like `KubernetesConnectionConfiguration` and use it as field in the finders. But it will break existing API for IpFinder. So I introduced the interface to track common configuration params.

Reply:
This interface extends our public API. Every new public API is a future liability and should be considered carefully. In this case there is no value for the users, so let's remove it.

Reply:
I can move this interface to package with finders and make it package-private, isn't it enough? Public API for IpFinder isn't changed, it still have the same setters. I'm agree that this interface doesn't look good but it's a limitation of exisiting IpFinder. Alternatives are:
1. Remove this interface and track k8s configuration params only with constructor of `KubernetesServiceAddressResolver`;
2. Provide a new object like `KubernetesConnectionKonfiguration` instead of interface, provide new IpFinder constructor that accept the configuration, deprecate setters in the IpFinder in favor of the constructor.

I like the second option. But it lead change of the IpFinder API (add new, deprecate old), is it ok?

Reply:
I've implemented the second option. I think, it got prettier.